src/index.js

```diff
@@ -165,89 +165,105 @@ exports.createFromPubKey = function (key, callback) {
     throw new Error('callback is required')
   }
 
-  let buf = key
-  if (typeof buf === 'string') {
-    buf = Buffer.from(key, 'base64')
-  }
+  try {
+    let buf = key
+    if (typeof buf === 'string') {
+      buf = Buffer.from(key, 'base64')
+    }
 
-  const pubKey = crypto.keys.unmarshalPublicKey(buf)
+    if (!Buffer.isBuffer(buf)) throw new Error('Supplied key is neither a base64 string nor a buffer')
 
-  pubKey.hash((err, digest) => {
-    if (err) {
-      return callback(err)
-    }
+    const pubKey = crypto.keys.unmarshalPublicKey(buf)
 
-    callback(null, new PeerId(digest, null, pubKey))
-  })
+    pubKey.hash((err, digest) => {
+      if (err) {
+        return callback(err)
+      }
+
+      callback(null, new PeerId(digest, null, pubKey))
```

Comment:
This is an anti-pattern, you should not call a callback from within a try/catch block because any future error that is thrown will circle back to this try catch.

Reply:
You need to move the callback to outside of the try/catch block